integration-test/request-blocking.spec.js

```diff
@@ -6,6 +6,21 @@ import { TEST_SERVER_ORIGIN } from './helpers/testPages'
 const testHost = 'privacy-test-pages.site'
 const testSite = `https://${testHost}/privacy-protections/request-blocking/`
 
+async function forDynamicRulesLoaded (backgroundPage) {
```

Comment:
I think this should have DNR in the name, like `forDynamicDNRRulesLoaded` and live in `helpers/backgroundWait.js`.

This is kind of a hack, but if it's good enough for now fair enough:
1. I can't see how better to do it without some fairly large changes.
2. The related helpers use similar hacks at the moment.
3. So long as we use this helper in the tests, changing how it works in the future won't require messing with the tests anyway.

Reply:
_(Also see this comment https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/2309/files#r1537480355, I had an idea for another way to do this.)_

Reply:
Done and moved (https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/2309/commits/9d43936ea5712fe31efc1c8643e225b92ecbbc60)